### ISSUE TITLE
heapless does not build for riscv32imc-unknown-none-elf target

### ISSUE THREAD

**Issue by rjoleary** [japaric/heapless#123]
rv32imc does not have atomics leading to this issue:

```
error[E0599]: no method named `compare_exchange_weak` found for type `core::sync::atomic::AtomicPtr<pool::Node<T>>` in the current scope
   --> /home/ryan/.cargo/registry/src/github.com-1ecc6299db9ec823/heapless-0.4.4/src/pool/mod.rs:278:33
    |
278 |                 match self.head.compare_exchange_weak(
    |                                 ^^^^^^^^^^^^^^^^^^^^^ method not found in `core::sync::atomic::AtomicPtr<pool::Node<T>>`
```

**Comment by jamesmunns** [comment]
I believe this would be closed by #120.

**Comment by japaric** [comment]
> I believe this would be closed by #120.

That PR is not sufficient. That target does not have atomic *loads* either so `AtomicUsize` et al. are missing from `core` meaning that the `spsc` module doesn't compile for this target even after disabling the "cas" Cargo feature.
